src/Query/DateParser.ts

```diff
@@ -75,12 +93,74 @@ export class DateParser {
             const unitOfTime = range === 'week' ? 'isoWeek' : (range as moment.unitOfTime.DurationConstructor);
             dateRange = [dateRange[0].startOf(unitOfTime), dateRange[1].endOf(unitOfTime)];
 
-            // Dates shall be at midnight eg 00:00
-            dateRange.forEach((d) => d.startOf('day'));
+            DateParser.setDateRangeToStartOfDay(dateRange);
 
             return dateRange;
         }
 
         return undefined;
     }
+
+    private static specificYearRegex = /[0-9]{4}/;
+    private static specificQuarterRegex = /[0-9]{4}-Q[1-4]/;
+    private static specificMonthRegex = /[0-9]{4}-[0-9]{2}/;
+    private static specificWeekRegex = /[0-9]{4}-W[0-9]{2}/;
+    private static specificYearFormat = 'YYYY';
+    private static specificQuarterFormat = 'YYYY-Q';
+    private static specificMonthFormat = 'YYYY-MM';
+    private static specificWeekFormat = 'YYYY-WW';
+
+    private static parseSpecificDateRange(input: string): [moment.Moment, moment.Moment] | undefined {
+        const yearMatch = input.match(DateParser.specificYearRegex);
+        if (yearMatch && yearMatch.length === 1 && yearMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(yearMatch[0], DateParser.specificYearFormat);
+        }
+
+        const quarterMatch = input.match(DateParser.specificQuarterRegex);
+        if (quarterMatch && quarterMatch.length === 1 && quarterMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(quarterMatch[0], DateParser.specificQuarterFormat);
+        }
+
+        const monthMatch = input.match(DateParser.specificMonthRegex);
+        if (monthMatch && monthMatch.length === 1 && monthMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(monthMatch[0], DateParser.specificMonthFormat);
+        }
+
+        const weekMatch = input.match(DateParser.specificWeekRegex);
+        if (weekMatch && weekMatch.length === 1 && weekMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(weekMatch[0], DateParser.specificWeekFormat);
+        }
```

Comment:
This is rather repetitious, and prone to errors in copying and pasting.

Can you perhaps build a container that you can loop over, something like:

```ts
{
    { DateParser.specificYearRegex, DateParser.specificYearFormat },
    { DatePArser.specificQuarterRegex, DateParser.specificQuarterFormat },
    ...
}
```

Reply:
Or better still, add the `format` value to be passed in to `buildSpecificDateRange()`, to save it having to switch on the format value...

```ts
{
    { DateParser.specificYearRegex, DateParser.specificYearFormat, 'year' },
    { DatePArser.specificQuarterRegex, DateParser.specificQuarterFormat, 'quarter' },
    ...
}
```

Reply:
In fact, I bet you could inline all those constants on lines 104 to 111.

Reply:
why are you using `{ ... }` and not `[ ... ]`? Or am I getting something wrong?

Reply:
Because I was typing pseudocode and didn’t notice.  Hence `something like` 🤣

Reply:
ah lol =) actually those brackets made me think about a solution I finally took for implementation =)